### PR TITLE
Flexible aws

### PR DESCRIPTION
**Pull request by chris-salemove**
Update AWS support:
 * Support provisioning a cluster in a pre-existing VPC/subnet
 * Support specifying master IP
 * Support specifying SERVICE_CLUSTER_IP_RANGE

### CODE REVIEW

File: cluster/aws/util.sh [hunk 1]
```diff
@@ -53,9 +53,7 @@ AWS_CMD="aws --output json ec2"
 AWS_ELB_CMD="aws --output json elb"
 AWS_ASG_CMD="aws --output json autoscaling"
 
-INTERNAL_IP_BASE=172.20.0
```
Comment (indrekj) [inline cluster/aws/util.sh]: do we want to get these changes into main `kubernetes` project also? If so, then I guess here should be something like  `if defined ? this one : 127.20.0`

Comment (chris-salemove) [inline cluster/aws/util.sh]: INTERNAL_IP_BASE wasn't removed entirely - the value is specified in config-default.sh which checks for override.

Comment (indrekj) [inline cluster/aws/util.sh]: oh right :)

**Comment by indrekj** [comment]
+1

**Comment by madis** [comment]
I think there should be better way for making changes to the kubernetes source then just merging to the fork's master. If the changes are suitable (with chance of getting merged) to be made to kubernetes then pull request should be created for the upstream. If not then different approach using existing kubernetes code & apis should be used.

One way would be to create pull requests to this fork but on a different branch, e.g. something like *sm_master* or *faster_master*. Then also creating pull request to upstream. Otherwise our custom changes keep piling up and at over time syncing with kubernetes original upstream will be come very difficult.

Another would be to push changes here, create pull request to upstream (google's repo) and work from feature branch here. This works as long as there is 1 non-merged feature branch at a time. Otherwise it will become the first option.

**Comment by madis** [comment]
I'm :+1: too after deciding how we will keep the fork in sync with upstream.

**Comment by stenver** [comment]
+1 to @madis concerns

**Comment by deiwin** [comment]
[off topic] screen shot 2015-10-14 at 10 16 31 pm

Probably need to think of a better way to define a PR peer reviewed than a comment starting with a '+1'

**Comment by jzd131** [comment]
@deiwin perhaps we can also add "security checked" or something like that as well

**Comment by paniagua** [comment]
@jzd131 what does the security checked would mean?

**Comment by stenver** [comment]
I think its one of the steps here: https://docsend.com/view/3dti596

Although I am not sure who wrote this doc, since it doesn't really reflect the reality

**Comment by jzd131** [comment]
@paniagua who ever does the code review could also look for security issues, or we need someone else to look at the code for security

**Comment by chris-salemove** [comment]
Gentlemen: please bring the +1/security discussion to a different thread. Way off-topic here.

**Comment by chris-salemove** [comment]
@madis @indrekj Agree we need to settle on a merge strategy.
One point to consider is that this is a fork of the 1.0.6 tag. I don't know whether we want to start merging with unreleased versions. Let's discuss in person.

**Comment by chris-salemove** [comment]
Closing in favor of new PR #2 for sm-master branch.

**Pull request closed without merging**